### PR TITLE
Add pull request template

### LINKED ISSUE #33: Create a PULL_REQUEST_TEMPLATE
#### Issue Summary

We would like to have a template for describing what has changed in the pull request being submitted.

[Github docs on this](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/) 

You can use [this file](https://github.com/sendgrid/open-source-library-data-collector/blob/master/.github/PULL_REQUEST_TEMPLATE.md) as an example

### PR DESCRIPTION
**Pull request by jeffgerlach**
Add in example template from https://github.com/sendgrid/open-source-library-data-collector/blob/master/.github/PULL_REQUEST_TEMPLATE.md and updated for this project. First time contributing to open source, let me know if I missed anything! Closes #33

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/nodejs-http-client?pullRequest=34) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/nodejs-http-client?pullRequest=34) before we can accept your contribution.<br/><hr/>**Jeff Gerlach** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/nodejs-http-client?pullRequest=34) it.</sub>

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+- [ ] Code passes all existing [tests](https://github.com/sendgrid/nodejs-http-client/tree/master/test)
+- [ ] Any new functionality added includes new unit tests in [`test/test.js`](https://github.com/sendgrid/nodejs-http-client/tree/master/test/test.js)
+- [ ] Create or update example code to show the new functionality in action.
+- [ ] All code, branch, and git naming and style conventions are followed (see [`CONTRIBUTING.md`](https://github.com/sendgrid/nodejs-http-client/blob/master/CONTRIBUTING.md#style-guidelines--naming-conventions))
+- [ ] Feature branch has been rebased off of the latest `master` branch. ( see [`CONTRIBUTING.md`](https://github.com/sendgrid/nodejs-http-client/blob/master/CONTRIBUTING.md#creating-a-pull-request) ).
+
+If you have questions, please send an email [Sendgrid](mailto:dx@sendgrid.com), or file a Github Issue in this repository.
+
+<!--
```
Comment (mbernier) [inline .github/PULL_REQUEST_TEMPLATE]: Can you please remove this section? This is linking back to sendgrid/docs and may confuse people. I removed it from the source file too - sorry about that!

**Comment by mbernier** [comment]
@jeffgerlach you did amazing! I made a mistake in my request, so I asked if you can make one small change for me. Please let me know if you have any issues and I will help out! 

Thank you so much for making one of your first PRs with us!! We hope you learned something and enjoyed the experience (also once we get this merged, we'll send you a link to request a t-shirt!!)

**Comment by jeffgerlach** [comment]
Removed the section as you requested, let me know if there's anything else you'd like changed!

**Comment by thinkingserious** [comment]
Hello @jeffgerlach,<br /><br/>Thanks again for the PR!<br />                                              <br />We want to show our appreciation by sending                                              you some swag. Could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by mbernier** [comment]
Thanks @jeffgerlach!!

**Pull request merged**